core/deadline.go

```diff
@@ -87,8 +90,20 @@ func NewDeadliner(ctx context.Context, deadlineFunc func(Duty) time.Time) *Deadl
 			case <-ctx.Done():
 				return
 			case duty := <-d.dutyChan:
+				deadline := deadlineFunc(duty)
+
+				// Ignore expired duties.
+				if deadline.Before(time.Now()) {
+					d.addedChan <- false
+					return
```

Comment:
replace return with continue with return. return will exit from this goroutine while we still want to add duties